rm-datasource/src/main/java/io/seata/rm/datasource/sql/struct/TableRecords.java

```diff
@@ -259,6 +262,75 @@ public static TableRecords buildRecords(TableMeta tmeta, ResultSet resultSet) th
         return records;
     }
 
+    /**
+     * check if the column is null and return
+     *
+     * @param tmeta the table meta
+     * @param colName the column nmae
+     */
+    private static ColumnMeta checkAndGetColumnMeta(TableMeta tmeta , String colName) throws SQLException {
+        ColumnMeta col = tmeta.getColumnMeta(colName);
+        if (col == null) {
+            throw new TableMetaException(colName, tmeta, true);
+        }
+        return col;
+    }
+
+
+    /**
+     * Build records table records.
+     *
+     * @param tmeta     the tmeta
+     * @param resultSet the result set
+     * @param statementProxy the statement proxy
+     * @return the table records
+     * @throws SQLException the sql exception
+     */
+    public static TableRecords buildRecords(TableMeta tmeta, ResultSet resultSet, StatementProxy statementProxy)
+        throws SQLException {
+        try {
+            return buildRecords(tmeta, resultSet);
+        } catch (TableMetaException e) {
```

Comment:
TableMetaException should be lightweight and contain only the necessary information.

Reply:
Do you mean ‘refreshable’ shouldn't be inside the TableMetaException?

Reply:
At  least, tableMeta information should not be included